apps/src/code-studio/components/libraries/LibraryClientApi.js

```diff
@@ -78,53 +77,18 @@ export default class LibraryClientApi {
   }
 
   async getClassLibraries(onSuccess, onError) {
-    getUserSections(sections => {
-      // TODO: Add backend controller action so this doesn't require multiple AJAX requests or this super clunky promise structure.
-      let requests = [];
-      let allLibraries = [];
-      let promises = [];
-      let libraryIds = {};
-      sections.forEach(section => {
-        requests.push(
-          $.ajax({
-            url: `/dashboardapi/v1/projects/section/${section.id}`,
-            method: 'GET',
-            dataType: 'json'
-          })
-        );
-      });
-      requests.forEach(request => {
-        promises.push(
-          new Promise((resolve, reject) => {
-            $.when(request)
-              .done(data => {
-                if (data) {
-                  let libraries = data
-                    .filter(library => !!library.libraryName)
-                    .map(library => {
-                      library.name = library.libraryName;
-                      library.description = library.libraryDescription;
-                      return library;
-                    });
-                  libraries.forEach(library => {
-                    if (!libraryIds[library.channel]) {
-                      allLibraries.push(library);
-                      libraryIds[library.channel] = true;
-                    }
-                  });
-                }
-                resolve();
-              })
-              .fail(error => {
-                console.warn('Error finding class libraries: ' + error);
-                reject();
-              });
-          })
-        );
-      });
-      Promise.all(promises).then(() => {
-        onSuccess(allLibraries);
-      });
-    });
+    let data;
+    try {
+      let response = await fetch('/api/libraries/section/all', {method: 'GET'});
```

Comment:
Does this need a section id?

Reply:
Ohh... I kept reading. Now I see what you're doing. So these are really libraries that are associated with a specific user through their enrollment in sections.